### PR TITLE
Series.sum optimization

### PR DESCRIPTION
**Pull request by 1e-to**
image

### CODE REVIEW

**Comment by AlexanderKalistratov** [comment]
@1e-to tests for bool series fails. We need to implement sum/nansum (and all other functions) for bools too.

**Comment by densmirn** [comment]
> Wouldn't you like to move implementations of `sum` and `nansum` from overloads and call it there?

The implementations looks the same.

**Comment by AlexanderKalistratov** [comment]
> Wouldn't you like to move implementations of `sum` and `nansum` from overloads and call it there?

didn't get what do you want to do

**Comment by densmirn** [comment]
> > Wouldn't you like to move implementations of `sum` and `nansum` from overloads and call it there?
> 
> didn't get what do you want to do

I meant something like this:
```python
def gen_sum_number_impl():
    """Generate sum number implementation."""
    def _sum_number_impl(self):
        length = len(self)
        result = 0
        for i in prange(length):
            if self[i]:
                result += self[i]

        return result

    return _sum_number_impl


@sdc_overload(sum)
def sdc_sum_overload(self):
    # ...
    if isinstance(dtype, (types.Boolean, bool)):
        return gen_sum_number_impl()
    # ...


@sdc_overload(nansum)
def sdc_nansum_overload(self):
    # ...
    if isinstance(dtype, (types.Boolean, bool)):
        return gen_sum_number_impl()
    # ...
```

File: sdc/functions/numpy_like.py [hunk 1]
```diff
@@ -201,6 +201,20 @@ def sdc_isnan_float_impl(self):
     ty_checker.raise_exc(dtype, 'int or float', 'self.dtype')
 
 
+def gen_sum_bool_impl():
+    """Generate sum bool implementation."""
+    def _sum_bool_impl(self):
+        length = len(self)
+        result = 0
+        for i in prange(length):
+            if self[i]:
```
Comment (AlexanderKalistratov) [inline sdc/functions/numpy_like.py]: Why do you need this `if self[i]` condition?

Comment (1e-to) [reply sdc/functions/numpy_like.py]: Without it also works, fixed

**Pull request merged**